go.mod

```diff
@@ -34,6 +35,7 @@ require (
 	github.com/prometheus/client_golang v1.8.0
 	github.com/prometheus/client_model v0.2.0
 	github.com/prometheus/common v0.14.0
+	github.com/smartystreets/goconvey v1.6.4
```

Comment:
@yaron2 I think there was a specific reason as to not use any specific go tools for testing. correct?

Reply:
We don't use custom test libraries.

Reply:
ok, i'll remove it, there are also use bou.ke/monkey library,that's ok?

Reply:
I like the patching library you are using for tests. I am OK to try it out since it would unlock a lot more testing, specially in the CLI code.

Reply:
/cc @yaron2  any concerns over adding this library for mocking functions?

Reply:
No concern.